tests/unit/DateTimeTest.php

```diff
@@ -0,0 +1,163 @@
+    {
+        return array(
+            array("2014-04-26 13:00:00", "2014-04-26 13:00:00", 'just now'),
+            array("2014-04-26 13:00:00", "2014-04-26 13:00:05", '5 seconds from now'),
+            array("2014-04-26 13:00:00", "2014-04-26 12:59:00", '1 minute ago'),
+            array("2014-04-26 13:00:00", "2014-04-26 12:45:00", '15 minutes ago'),
+            array("2014-04-26 13:00:00", "2014-04-26 13:15:00", '15 minutes from now'),
+            array("2014-04-26 13:00:00", "2014-04-26 14:00:00", '1 hour from now'),
+            array("2014-04-26 13:00:00", "2014-04-26 15:00:00", '2 hours from now'),
+            array("2014-04-26 13:00:00", "2014-04-26 12:00:00", '1 hour ago'),
+            array("2014-04-26", "2014-04-25", '1 day ago'),
+            array("2014-04-26", "2014-04-24", '2 days ago'),
+            array("2014-04-26", "2014-04-28", '2 days from now'),
+            array("2014-04-01", "2014-04-15", '2 weeks from now'),
+            array("2014-04-15", "2014-04-07", '1 week ago'),
+            array("2014-01-01", "2014-04-01", '3 months from now'),
+            array("2014-05-01", "2014-04-01", '1 month ago'),
+            array("2015-05-01", "2014-04-01", '1 year ago'),
+            array("2014-05-01", "2016-04-01", '2 years from now'),
+
+            array("2014-04-26 13:00:00", "2014-04-26 13:00:00", 'w tym momencie', 'pl'),
+            array("2014-04-26 13:00:00", "2014-04-26 13:00:05", 'za 5 sekund', 'pl'),
+            array("2014-04-26 13:00:00", "2014-04-26 12:59:00", 'minutę temu', 'pl'),
+            array("2014-04-26 13:00:00", "2014-04-26 12:45:00", '15 minut temu', 'pl'),
+            array("2014-04-26 13:00:00", "2014-04-26 13:15:00", 'za 15 minut', 'pl'),
+            array("2014-04-26 13:00:00", "2014-04-26 14:00:00", 'za godzinę', 'pl'),
+            array("2014-04-26 13:00:00", "2014-04-26 15:00:00", 'za 2 godziny', 'pl'),
+            array("2014-04-26 13:00:00", "2014-04-26 12:00:00", 'godzinę temu', 'pl'),
+            array("2014-04-26 13:00:00", "2014-04-26 15:00:00", 'za 2 godziny', 'pl'),
+            array("2014-04-26 13:00:00", "2014-04-26 12:00:00", 'godzinę temu', 'pl'),
+            array("2014-04-26", "2014-04-25", 'wczoraj', 'pl'),
+            array("2014-04-26", "2014-04-24", '2 dni temu', 'pl'),
+            array("2014-04-26", "2014-04-28", 'za 2 dni', 'pl'),
+            array("2014-04-01", "2014-04-15", 'za 2 tygodnie', 'pl'),
+            array("2014-04-15", "2014-04-07", 'tydzień temu', 'pl'),
+            array("2014-01-01", "2014-04-01", 'za 3 miesiące', 'pl'),
+            array("2014-05-01", "2014-04-01", 'miesiąc temu', 'pl'),
+            array("2015-05-01", "2014-04-01", 'rok temu', 'pl'),
+            array("2014-05-01", "2016-04-01", 'za 2 lata', 'pl'),
+            array("2014-05-01", "2009-04-01", '5 lat temu', 'pl'),
+        );
+    }
+
+    /**
+     * @return array
+     */
+    public function preciseDifferenceDataProvider()
+    {
+        return array(
```

Comment:
I could, I thought it was easier and DRYer this way, simply add a new block to the dataprovider and you're good to go, the test code stays the same.